### PR TITLE
support for ebs optimization in launch templates

### PR DESCRIPTION
**Pull request by jrowinski3d**
## what
* Enabled EBS Optimization in the launch template.

## why
* Give the user ability to enable ebs optimization, default will be false. Of course the instance_type will need to be capable of doing so.

## references
* closes [#123](https://github.com/cloudposse/terraform-aws-eks-node-group/issues/97)

### CODE REVIEW

File: launch-template.tf [hunk 1]
```diff
@@ -25,6 +25,7 @@ locals {
   launch_template_configured = length(var.launch_template_id) == 1
   generate_launch_template   = local.enabled && local.launch_template_configured == false
   fetch_launch_template      = local.enabled && local.launch_template_configured
+  #ebs_optimized              = local.enabled && local.launch_template_configured
```
Comment (aknysh) [inline launch-template.tf:28]: why do we need this comment?

File: launch-template.tf [hunk 2]
```diff
@@ -50,6 +51,8 @@ resource "aws_launch_template" "default" {
 
   count = local.generate_launch_template ? 1 : 0
 
+  ebs_optimized = var.ebs_optimized ? true : false
```
Comment (aknysh) [inline launch-template.tf:54]:
```suggestion
  ebs_optimized = var.ebs_optimized
```

File: variables.tf [hunk 3]
```diff
@@ -258,6 +258,12 @@ variable "module_depends_on" {
   description = "Can be any value desired. Module will wait for this value to be computed before creating node group."
 }
 
+variable "ebs_optimized" {
+  type        = bool
+  default     = false
+  description = "Whether or not to launch instances with ebs_optimization. Default false"
```
Comment (aknysh) [inline variables.tf:264]:
```suggestion
  description = "Whether or not to launch instances with EBS optimization. Default is `false`"
```

**Review by aknysh — COMMENTED** [review]
@jrowinski3d thank you
please see comments

**Comment by jrowinski3d** [comment]
> @jrowinski3d thank you please see comments

Thank you, sorry missed those. I updated to your recommendations.

**Review by aknysh — APPROVED** [review]

**Pull request merged**